### PR TITLE
Regroup equipmentDeletion notifications

### PR DESCRIPTION
**Pull request by ayolab**
Signed-off-by: Ayoub LABIDI <ayoub.labidi@protonmail.com>

### CODE REVIEW

File: src/main/java/org/gridsuite/study/server/notification/EquipmentDeletionNotification.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package org.gridsuite.study.server.notification;
```
Comment (anistouri) [inline src/main/java/org/gridsuite/study/server/notification/EquipmentDeletionNotification.java:1]: Add Copyright

File: src/main/java/org/gridsuite/study/server/service/StudyService.java [hunk 2]
```diff
@@ -1415,30 +1417,24 @@ private void updateStatuses(UUID studyUuid, UUID nodeUuid, List<ModificationInfo
...
-                deletions.forEach(modif -> notificationService.emitStudyEquipmentDeleted(studyUuid, nodeUuid, NotificationService.UPDATE_TYPE_STUDY,
-                        allImpactedSubstationIds, modif.getEquipmentType(), modif.getEquipmentId()));
-                updateStatuses(studyUuid, nodeUuid);
-                break;
-            }
-            default: {
-                Set<String> substationIds = getSubstationIds(modifications);
-                notificationService.emitStudyChanged(studyUuid, nodeUuid, NotificationService.UPDATE_TYPE_STUDY, substationIds);
+
+                Set<String> impactedSubstationIds = getSubstationIds(modifications).stream()
```
Comment (laurgarn) [inline src/main/java/org/gridsuite/study/server/service/StudyService.java:1426]: May be collect all impacted substations id, all suppressed substations id, then use CollectionUtils.substract with the two collections. That could be a little more explicit.

**Comment by laurgarn** [comment]
Tested OK

File: src/main/java/org/gridsuite/study/server/notification/NotificationService.java [hunk 3]
```diff
@@ -4,7 +4,7 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
  */
-package org.gridsuite.study.server.service;
```
Comment (etiennehomer) [inline src/main/java/org/gridsuite/study/server/notification/NotificationService.java:7]: Keep NotificationService in service package nope ?

File: src/main/java/org/gridsuite/study/server/notification/payload/EquipmentDeletionNotification.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
+/**
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package org.gridsuite.study.server.notification.payload;
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/server/notification/payload/EquipmentDeletionNotification.java:8]: Rename payload by dto

File: src/main/java/org/gridsuite/study/server/notification/payload/EquipmentDeletionNotification.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.Setter;
+
+@AllArgsConstructor
+@Setter
+@Getter
+public class EquipmentDeletionNotification {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/server/notification/payload/EquipmentDeletionNotification.java:17]: Why we do not use the class `EquipmentDeletionInfos`

File: src/main/java/org/gridsuite/study/server/notification/payload/NetworkImpcatsNotificationPayload.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+@AllArgsConstructor
+@Setter
+@Getter
+@Builder
+@NoArgsConstructor
+public class NetworkImpcatsNotificationPayload {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/server/notification/payload/NetworkImpcatsNotificationPayload.java:23]: Renam by `NetworkImpcatsInfos`

Comment (laurgarn) [reply src/main/java/org/gridsuite/study/server/notification/payload/NetworkImpcatsNotificationPayload.java:23]: Uh, NetworkImp**ac**tsInfos, I'd say

File: src/main/java/org/gridsuite/study/server/service/StudyService.java [hunk 7]
```diff
@@ -1447,42 +1450,38 @@ private void updateStatuses(UUID studyUuid, UUID nodeUuid, List<ModificationInfo
...
-                            deletionInfo.getEquipmentType(), deletionInfo.getEquipmentId()));
-                updateStatuses(studyUuid, nodeUuid);
-                break;
-            }
-            case LINE_SPLIT_WITH_VOLTAGE_LEVEL: {
-                Set<String> allImpactedSubstationIds = modifications.stream()
-                        .map(ModificationInfos::getSubstationIds).flatMap(Set::stream).collect(Collectors.toSet());
-                List<EquipmentModificationInfos> deletions = modifications.stream()
+            default: {
+                Set<EquipmentModificationInfos> deletions = modifications.stream()
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/server/service/StudyService.java:1466]: Why `Set` ?
If you want to remove duplications add `@EqualsAndHashCode` to the `EquipmentDeletionNotification` class

                Set<EquipmentDeletionNotification> deletionsInfos = modifications.stream()
                    .filter(modif -> modif.getType() == ModificationType.EQUIPMENT_DELETION)
                    .map(EquipmentDeletionInfos.class::cast)
                    .map(d -> new EquipmentDeletionNotification(d.getEquipmentId(), d.getEquipmentType()))
                    .collect(Collectors.toSet());

                // we need to filter out the substation ids that are deleted
                Set<String> deletionsIds = deletionsInfos.stream().map(EquipmentDeletionNotification::getId).collect(Collectors.toSet());
                Set<String> impactedSubstationIds = getSubstationIds(modifications).stream()
                    .filter(id -> !deletionsIds.contains(id))
                    .collect(Collectors.toSet());

                notificationService.emitStudyChanged(studyUuid, nodeUuid, NotificationService.UPDATE_TYPE_STUDY,
                    NetworkImpcatsNotificationPayload
                        .builder()
                        .impactedSubstationsIds(impactedSubstationIds)
                        .deletedEquipments(deletionsInfos)
                        .build());
                updateStatuses(studyUuid, nodeUuid);

File: src/test/java/org/gridsuite/study/server/StudyTest.java [hunk 8]
```diff
@@ -1157,16 +1159,14 @@ private void checkUpdateNodesMessageReceived(UUID studyUuid, List<UUID> nodesUui
         assertEquals(NotificationService.NODE_UPDATED, headersStatus.get(NotificationService.HEADER_UPDATE_TYPE));
     }
 
-    private void checkEquipmentMessagesReceived(UUID studyNameUserIdUuid, UUID nodeUuid, String headerUpdateTypeId,
-            Set<String> modifiedIdsSet) {
+    private void checkEquipmentMessagesReceived(UUID studyNameUserIdUuid, UUID nodeUuid, NetworkImpcatsNotificationPayload expectedPayload) throws Exception {
         // assert that the broker message has been sent for updating study type
         Message<byte[]> messageStudyUpdate = output.receive(TIMEOUT, studyUpdateDestination);
-        assertEquals("", new String(messageStudyUpdate.getPayload()));
+        assertEquals(mapper.writeValueAsString(expectedPayload), new String(messageStudyUpdate.getPayload()));
```
Comment (SlimaneAmar) [inline src/test/java/org/gridsuite/study/server/StudyTest.java:1165]: Use the Json matcher

File: src/test/java/org/gridsuite/study/server/NetworkModificationTest.java [hunk 9]
```diff
@@ -2463,16 +2490,14 @@ private void checkUpdateEquipmentModificationMessagesReceived(UUID studyNameUser
     }
 
     private void checkEquipmentMessagesReceived(UUID studyNameUserIdUuid, UUID nodeUuid, String headerUpdateTypeId,
-            Set<String> modifiedIdsSet) {
+            NetworkImpcatsNotificationPayload expectedPayload) throws Exception {
         // assert that the broker message has been sent for updating study type
         Message<byte[]> messageStudyUpdate = output.receive(TIMEOUT, studyUpdateDestination);
-        assertEquals("", new String(messageStudyUpdate.getPayload()));
+        assertEquals(mapper.writeValueAsString(expectedPayload), new String(messageStudyUpdate.getPayload()));
```
Comment (SlimaneAmar) [inline src/test/java/org/gridsuite/study/server/NetworkModificationTest.java:2496]: Use the Json matcher

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

**Review by SlimaneAmar — APPROVED** [review]

**Pull request merged**